include/nanobind/intrusive/ref.h

```diff
@@ -136,6 +136,11 @@ template <typename T> struct type_caster<nanobind::ref<T>> {
 
     static handle from_cpp(const ref<T> &value, rv_policy policy,
                            cleanup_list *cleanup) noexcept {
+        if constexpr (std::is_base_of_v<intrusive_base, T>)
+            if (policy != rv_policy::copy && value.get())
```

Comment:
I think that we might want to exclude `rv_policy::move` as well

Reply:
I've done so.